frontend/src/pages/modelRegistry/screens/RegisterModel/utils.ts

```diff
@@ -140,12 +141,19 @@ const isSubmitDisabledForCommonFields = (formData: RegistrationCommonFormData):
   );
 };
 
-export const isRegisterModelSubmitDisabled = (formData: RegisterModelFormData): boolean =>
+export const isRegisterModelSubmitDisabled = (
+  formData: RegisterModelFormData,
+  registeredModels: RegisteredModelList,
+): boolean =>
   !formData.modelName ||
   isSubmitDisabledForCommonFields(formData) ||
-  !isNameValid(formData.modelName);
+  !isNameValid(formData.modelName) ||
+  isModelNameExisting(formData.modelName, registeredModels);
 
 export const isRegisterVersionSubmitDisabled = (formData: RegisterVersionFormData): boolean =>
   !formData.registeredModelId || isSubmitDisabledForCommonFields(formData);
 
 export const isNameValid = (name: string): boolean => name.length <= MR_CHARACTER_LIMIT;
+
+export const isModelNameExisting = (name: string, registeredModels: RegisteredModelList): boolean =>
```

Comment:
small nit: I think we should have a unit test for this function. Ideally we should have one for `isNameValid` too - but since we don't.. that's out of scope of this PR.

Reply:
make mentioned this being in a tech debt list and that we'll end up refactoring form validation logic as part of a larger dashboard UX consistency effort

Reply:
That may be a bad excuse to prevent writing a test though. I think @manaswinidas is right - it'd probably be a good start to create a unit tests file for this utils file even if we just put this one small test in there for now.

Reply:
I'm skeptical we'll get to that form validation refactor any time soon, we should probably not use that as a reason to not write tests. Sorry for changing my position on that @gitdallas ... @manaswinidas keeping us to high standards here 😄